location/java/android/location/LocationManager.java

```diff
@@ -1725,6 +1725,11 @@ public void removeUpdates(@NonNull PendingIntent pendingIntent) {
     public boolean hasProvider(@NonNull String provider) {
         Preconditions.checkArgument(provider != null, "invalid null provider");
 
+        if (NETWORK_PROVIDER.equals(provider)) {
+            android.util.Log.d("LocationManager", "returning false from hasProvider(NETWORK_PROVIDER)");
+            return false;
```

Comment:
Whilst not essential for official GrapheneOS releases, we should change this to check if there are any network location providers within the OS before we return false at some point.

Reply:
There can be only one "network" location provider (`NETWORK_PROVIDER` refers to the "network" string constant).
If it's present, this commit should be dropped. However, it's simple to make it so that this commit doesn't break the "real" network location provider, I'll update it now.